tests/test_synapse/test_synapse_factory.py

```diff
@@ -74,12 +74,26 @@ def test_synlocation_to_segx(self):
 
 
 def test_determine_synapse_type():
-    """Unit test for determine_synapse_type."""
-    assert SynapseFactory.determine_synapse_type(True, False) == SynapseType.GABAAB
-    assert SynapseFactory.determine_synapse_type(True, True) == SynapseType.GABAAB
-
-    # Test when synapse is excitatory with plasticity
-    assert SynapseFactory.determine_synapse_type(False, True) == SynapseType.GLUSYNAPSE
-
-    # Test when synapse is excitatory without plasticity
-    assert SynapseFactory.determine_synapse_type(False, False) == SynapseType.AMPANMDA
+    # Mocking a pd.Series for syn_description
+    syn_description = pd.Series({
+        SynapseProperty.TYPE: 50,
+        SynapseProperties.plasticity[0]: None,
+        SynapseProperties.plasticity[1]: float('nan')
+    })
+
+    # Test when synapse is inhibitory
+    assert SynapseFactory.determine_synapse_type(syn_description) == SynapseType.GABAAB
+
+    # Test when synapse is excitatory with incomplete plasticity
+    syn_description[SynapseProperty.TYPE] = 150
+    syn_description[SynapseProperties.plasticity[0]] = 1.0  # Just one property set
+    assert SynapseFactory.determine_synapse_type(syn_description) == SynapseType.AMPANMDA
+
+    # Test when synapse is excitatory with all plasticity properties filled
+    for prop in SynapseProperties.plasticity:
+        syn_description[prop] = 1.0  # Assign a valid value to each plasticity property
+    assert SynapseFactory.determine_synapse_type(syn_description) == SynapseType.GLUSYNAPSE
+
+    # Test when synapse is excitatory with one plasticity property as NaN
```

Comment:
this is the same test as in line 90, no?

Reply:
Actually it tests a slightly different case
- On line 90 the values are [150, 1.0]
- On line 99 the values are [150, nan, 1.0, 1.0, 1.0, 1.0, 1.0]
Both should evaluate to AMPANMDA.

Reply:
I thought on line 90 it was [1.0, nan, 150]?
Either way, if you say they are different, it's ok.

Reply:
Let me put a debugger and double check the exact values.

Reply:
I updated the implementation to raise an exception when the synapse type cannot be determined.

Reply:
Before it was assuming the synapse type to be AMPANMDA when some plasticity params (not all) were missing. That could lead to errors.